float64 cx1
float64 cy1
float64 cx2
float64 cy2
bool appear

